eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/threading/PollingThread.java

```diff
@@ -134,135 +89,39 @@ public PollingThread(
    * @since 1.0.0
    */
   public void start() {
-    // Check if the executor service is shutdown
-    checkShutdown();
-
     // Check if the thread is already running
-    if (isRunning()) {
-      throw new IllegalStateException("Polling thread is already running.");
+    if (isRegistered) {
+      throw new IllegalStateException("Polling thread is already registered.");
     }
 
     // Schedule the thread to run at the specified interval
-    scheduledFuture =
-        executorService.scheduleAtFixedRate(
-            this, pollingIntervalStartDelay, pollingInterval, pollingIntervalTimeUnit);
+    gatewayHook
+        .getGatewayContext()
+        .getExecutionManager()
+        .register(
+            IgnitionEwonConnectorHook.BUNDLE_PREFIX_EWON,
+            threadName,
+            this,
+            (int) pollingIntervalTimeUnit.toMillis(pollingInterval));
+    isRegistered = true;
   }
 
   /**
-   * Stops the polling thread using the specified interrupt behavior.
+   * Stops the polling thread by unregistering it from the execution manager.
    *
-   * @param mayInterruptIfRunning true if in-progress polls should be interrupted; otherwise,
-   *     in-progress polls are allowed to complete
-   * @throws IllegalStateException if the polling thread executor service has been shutdown.
    * @since 1.0.0
    */
-  public void stop(boolean mayInterruptIfRunning) {
-    // Check if the executor service is shutdown
-    checkShutdown();
-
+  public void stop() {
     // Cancel the thread (if started)
-    if (scheduledFuture != null) {
-      scheduledFuture.cancel(mayInterruptIfRunning);
+    if (isRegistered) {
+      gatewayHook
+          .getGatewayContext()
+          .getExecutionManager()
+          .unRegister(IgnitionEwonConnectorHook.BUNDLE_PREFIX_EWON, threadName);
```

Comment:
There is also a shutdown method. The limited docs state will gracefully shutdown.  Should this be called before unregister?

Reply:
This is the execution manager of Ignition gateway. Shutting it down would likely have disastrous results.